packages/mobile/src/screens/collection-screen/CollectionScreen.tsx

```diff
@@ -68,32 +68,12 @@ const useStyles = makeStyles(({ spacing }) => ({
  */
 export const CollectionScreen = () => {
   const { params } = useRoute<'Collection'>()
+  const { id, searchCollection, slug, collectionType } = params ?? {}
   const dispatch = useDispatch()
 
-  // params is incorrectly typed and can sometimes be undefined
-  const {
-    id: idParam,
-    searchCollection,
-    collectionName,
-    collectionType
-  } = params ?? {}
-
-  const id = useMemo(() => {
-    if (collectionName) {
-      // Use collectionName from params if provided
-      // This is to support deep linking
-      // TODO: update this when collections are updated to use slug url format
-      // https://linear.app/audius/issue/C-1198/update-mobile-deep-linking-to-support-collection-slug-url-format
-      const nameParts = collectionName.split('-')
-      const collectionId = parseInt(nameParts[nameParts.length - 1], 10)
-      return collectionId as number
-    }
-    return idParam as number
-  }, [collectionName, idParam])
-
   const handleFetchCollection = useCallback(() => {
-    dispatch(fetchCollection(id))
-  }, [dispatch, id])
+    dispatch(fetchCollection(id, slug && decodeURIComponent(slug)))
```

Comment:
why do we need to decode the slug here?

Reply:
Good question, so for urls that have special characters, they are typically encoded, so decoding here to get the right special characters back

Reply:
gotcha ty!